### PR TITLE
Import Synonyms from CSVs

### LINKED ISSUE #437: Allow importing of synonyms from CSV file
As discussed on Slack:

- synonyms CSV just contains synonym, table-name, name
- we look for the name in the that table. If there is only one, use that GUID for the synonym
- If there is more than one, then either:
- invite the user to select the entry to use (providing the uniqueness fields), and use the selected GUID
- stop parsing, indicate that it relates to multiple existing entities.

### PR DESCRIPTION
**Pull request by robintw**
This PR adds the ability to import Synonyms.

These are imported as part of the 'Import Metadata' function in Pepys Admin, and come from `Synonyms.csv`. The CSV file lists the synonym itself, the table it is synonym for, and the target_name - which is the name of the target in that table. If there are multiple entries with that name in the target table, then an interface is provided for the user to select which entry to link the synonym to.

Fixes #437

### CODE REVIEW

**Comment by IanMayo** [comment]
@robintw - you have a conflict to resolve too, please.  It looks nice & straightforward though.

File: pepys_import/utils/data_store_utils.py [hunk 1]
```diff
@@ -29,6 +34,106 @@ def import_from_csv(data_store, path, files, change_id):
...
+                    print("Skipping row")
+                    continue
+                else:
+                    guid = getattr(chosen_item, pri_key_column_name)
+                    data_store.add_to_synonyms(values["table"], values["synonym"], guid, change_id)
+
+
+def ask_user_for_synonym_link(data_store, results, values):
+    options = [
+        f"{result.name}: Nationality = {result.nationality_name}, Identifier = {result.identifier}"
```
[the remark this answers is not in the document]
Comment (robintw) [reply pepys_import/utils/data_store_utils.py:106]: Ah, the response to both this issue and the other you raised is that @IanMayo agreed that we wouldn't be supporting synonyms for Sensor objects (see https://github.com/debrief/pepys-import/issues/387#issuecomment-638176544 - there was a discussion on Slack) - so the code only needs to support Synonyms for Platforms at the moment.

File: tests/sample_data/csv_files/for_synonym_tests/Synonyms.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+synonym,table,target_name
+PLATFORM-1-Synonym,Platforms,PLATFORM-1
+PLATFORM-1-Synonym2,Platforms,PLATFORM-1
+PLATFORM-2-Synonym,Platforms,PLATFORM-2
+PLATFORM-Duplicated-Synonym,Platforms,PLATFORM-DuplicatedName
```
[the remark this answers is not in the document]
Comment (robintw) [reply tests/sample_data/csv_files/for_synonym_tests/Synonyms.csv:5]: See response to previous issue: we're not supporting synonyms for Sensors, so this doesn't apply.

**Comment by robintw** [comment]
@BarisSari We could potentially add the Synonyms.csv file to the main CSV files folder, but there are two potential issues:

1. We'd have to add extra Platforms, and this would require changing quite a lot of tests to deal with different numbers of platforms coming out in the summary tables - and this means that if we do any changes to the Platforms.csv again to expand the tests for synonyms then we'd have to change all these tests again.

2. There is the potential for requiring user interaction to parse the Synonyms.csv file. It wouldn't be required with the Platforms.csv file that we use normally for the tests, but would be required if changes were made to the Platforms.csv - some of which are needed to do the tests needed for the synonyms import (as we need to test with a duplicate platform). This would then screw up all the tests that used the `populate_measurements` method as they would require mocking the user interaction. Basically, it'd break a load of tests and require more complicated mocking that isn't really needed.

Given those reasons, I think we should leave these with multiple copies - the files are all small, so there's not much cost to having duplicates. Obviously I'm biased as I wrote it this way to start with - so happy to change it if you think we should @BarisSari @IanMayo.

**Comment by IanMayo** [comment]
@robintw - your points seem reasonable, but I welcome @BarisSari thoughts.

File: pepys_import/utils/data_store_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BarisSari) [reply pepys_import/utils/data_store_utils.py:106]: Aah okay, I knew that I missed something :) Thanks for the answer.
